### PR TITLE
setting up express server

### PR DESCRIPTION
**Pull request by willemclarke**
dont know about cache folder?? will have to gitignore it or something

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -16,10 +13,8 @@
     "moduleResolution": "node",
     "resolveJsonModule": true,
     "isolatedModules": true,
-    "noEmit": true,
+    // "noEmit": true,
```
Comment (dbousamra) [inline tsconfig.json]: Remove this

File: .gitignore [hunk 2]
```diff
@@ -4,6 +4,7 @@
 /node_modules
 /.pnp
 .pnp.js
+dist
```
Comment (sbousamra) [inline .gitignore]: Add .cache to your .gitignore

File: package.json [hunk 3]
```diff
@@ -3,37 +3,31 @@
...
-    ],
-    "development": [
-      "last 1 chrome version",
-      "last 1 firefox version",
-      "last 1 safari version"
-    ]
+    "build:server": "tsc",
+    "build:client": "parcel build src/client/index.html --out-dir dist/client --no-minify",
+    "start": "node ./dist/server/index.js",
+    "develop:server:src": "tsc -w",
```
Comment (sbousamra) [inline package.json]: probably only want to watch the server folder so server doesn't restart if you make client changes

File: package.json [hunk 4]
```diff
@@ -3,37 +3,31 @@
...
+    "develop": "npm-run-all -p develop:server:src develop:server:dist develop:client"
   },
   "devDependencies": {
+    "@types/body-parser": "^1.17.1",
+    "@types/express": "^4.17.2",
     "@types/lodash": "^4.14.149",
     "@types/node": "^13.5.1",
     "@types/react": "^16.9.19",
     "@types/react-dom": "^16.9.5",
+    "less": "^3.10.3",
```
Comment (sbousamra) [inline package.json]: is less a dev dependency? Isn't it a requirement for your app to run?

File: src/client/index.html [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+  </head>
+
+  <body>
+    <div id="root"></div>
+    <script src="./index.tsx"></script>
```
Comment (sbousamra) [inline src/client/index.html]: why not just import the css here?

File: src/server/index.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import { Application, Request, Response } from "express";
+import * as _ from "lodash";
+const express = require("express");
```
Comment (sbousamra) [inline src/server/index.ts]: use import * as express from 'express', no need to use require

File: src/server/index.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+app.get("/api/", (req: Request, res: Response) => {
+  console.info(`This is an API request. Request path is: ${req.path}`);
+  res.status(200).json({
+    hello: "world"
+  });
+});
+
+app.use(express.static("dist/client"));
+
+(async () => {
```
Comment (sbousamra) [inline src/server/index.ts]: why use async here? No need for a promise here

File: src/server/index.ts (hunk 6 above)
Comment (RICHIEm8) [reply src/server/index.ts]: import * as express from 'express' was returning an error

File: src/server/index.ts (hunk 7 above)
Comment (sbousamra) [reply src/server/index.ts]: Also, why is it a function to app listen, why not just call app.listen directly

File: src/server/index.ts (hunk 6 above)
Comment (RICHIEm8) [reply src/server/index.ts]: > use import * as express from 'express', no need to use require

import * as express from 'express' was returning an error

**Pull request merged**